### PR TITLE
Feature/initial implementation

### PR DESCRIPTION
**Pull request by sitepark-veltrup**
See https://sitepark.github.io/atoolo-docs/develop/bundles/runtime-check/

### CODE REVIEW

File: src/Console/Command/CheckCommand.php [hunk 1]
```diff
@@ -18,15 +21,124 @@
...
 
+    protected function configure(): void
+    {
+        $this
+            ->setHelp('Command to performs a check of the runtime environment')
+            ->addOption(
+                'skip',
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                'Skip check for different areas. E.g. for "fpm" .',
```
Comment (sitepark-schaeper) [inline src/Console/Command/CheckCommand.php:38]: maybe we add a more detailed description on what exactly can be "skipped". afaik these can be scopes like 'fpm' or 'cli' or direct checkers like 'logger' or 'php'.

File: src/Service/Checker/CheckerCollection.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+     */
+    public function __construct(
+        private readonly iterable $checkers
+    ) {
+    }
+
+    /**
+     * @param array<string> $skip
+     */
+    public function check(array $skip): CheckStatus
```
Comment (sitepark-schaeper) [inline src/Service/Checker/CheckerCollection.php:22]: I think `$skip` may rather be called `$scopesToSkip` and should be optional.

**Review by sitepark-schaeper — COMMENTED** [review]
Just some minor details; looks generally good

File: src/Console/Command/CheckCommand.php [hunk 3]
```diff
@@ -18,15 +21,124 @@
...
+                'returns the exit code 1 if an error occurs',
+                true
+            )
+            ->addOption(
+                'json',
+                null,
+                InputOption::VALUE_NEGATABLE,
+                'output result in json.',
+                false
+            )
```
[the remark this answers is not in the document]
Comment (sitepark-veltrup) [reply src/Console/Command/CheckCommand.php:61]: I might change it again if there is another format.

**Comment by sitepark-veltrup** [comment]
The [optimization](https://github.com/sitepark/atoolo-runtime-check-bundle/pull/4/commits/0bba7c927c215d5862ff90319fba5d5c63bf0262) with `array_map` unfortunately does not work, as an index-based array is returned here if an index-based array is passed.

File: src/Console/Command/CheckCommand.php [hunk 4]
```diff
@@ -18,15 +21,124 @@
...
+        $this->outputResults(
+            $output,
+            $runtimeStatus,
+            $typedInput->getBoolOption('json')
+        );
+
+        $failOnError = $typedInput->getBoolOption('fail-on-error');
+        return $runtimeStatus->isSuccess() || !$failOnError
+                ? Command::SUCCESS
+                : Command::FAILURE;
```
[the remark this answers is not in the document]
Comment (sitepark-schaeper) [reply src/Console/Command/CheckCommand.php:98]: oh I see, my bad

File: src/Service/Checker/MonologChecker.php [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                . $this->maxLogDirSize . ' bytes';
+        }
+        if (
+            $this->maxLogFileRotations > 0
+            && ($reportData['logfile-rotations'] ?? 0)
+                > $this->maxLogFileRotations
+        ) {
+            $errors[] = 'logfile rotations exceed '
+                . $this->maxLogFileRotations;
+        }
```
[the remark this answers is not in the document]
Comment (sitepark-schaeper) [reply src/Service/Checker/MonologChecker.php:198]: okay - I guess this probably won't come up anyway

**Comment by sitepark-schaeper** [comment]
> The [optimization](https://github.com/sitepark/atoolo-runtime-check-bundle/pull/4/commits/0bba7c927c215d5862ff90319fba5d5c63bf0262) with `array_map` unfortunately does not work, as an index-based array is returned here if an index-based array is passed.

Yeah I make this mistake all the time...

**Review by sitepark-schaeper — APPROVED** [review]

**Pull request merged**